lib/EnzymeCore/src/EnzymeCore.jl

```diff
@@ -797,4 +798,92 @@ end
 
 Combined(mode::ReverseMode) = mode
 
+"""
+    ChunkStrategy
+
+Abstract type gathering strategies for chunk size selection.
+
+# Subtypes
+
+- [`SmallestChunk`](@ref)
+- [`LargestChunkk`](@ref)
+- [`FixedChunk`](@ref)
+- [`AutoChunk`](@ref)
+"""
+abstract type ChunkStrategy end
+
+"""
+    SmallestChunk()
+
+Select chunk size equal to 1, so that the corresponding array is processed in as many chunks as it has elements.
+
+!!! tip
+    In the current Enzyme interface, this strategy is equivalent to setting `chunk = nothing`.
+"""
+struct SmallestChunk <: ChunkStrategy end
+
+"""
+    LargestChunk()
+
+Select chunk size equal to the number of elements, so that the corresponding array is processed in a single chunk.
+"""
+struct LargestChunk <: ChunkStrategy end
+
+"""
+    FixedChunk{C}()
+
+Select chunk size equal to a fixed integer `C`.
+
+!!! tip
+    In the current Enzyme interface, this chunk strategy is equivalent to setting `chunk = Val(C)`.
+
+!!! warning
+    This chunk strategy will error if the corresponding array has length `< C`.
+"""
+struct FixedChunk{C} <: ChunkStrategy end
+
+"""
+    AutoChunk()
+
+Select chunk size automatically based on internal Enzyme-specific heuristics, which are subject to change.
+"""
+struct AutoChunk <: ChunkStrategy end
+
+const DEFAULT_CHUNK_SIZE = 16
```

Comment:
for type reasons, should we make this Val(16), otherwise it wont necessarily be a constant within pick_chunksize?